Appful/Token.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Token.swift
+//  Appful
+//
+//  Created by Karl Rivest Harnois on 16-04-27.
+//  Copyright © 2016 Dynamo. All rights reserved.
+//
+
+import Foundation
+
+//
+// API
+//
+
+public func fetchToken(email: String,
+                     _ password: String,
+                     _ completionHandler: JsonTaskCompletionHandler) -> NSURLSessionDataTask? {
+    return authenticationRequest(email, password).flatMap{ request in
```

Comment:
I struggled with understanding this use of `.flatMap` but I think I understand now. Essentially this is unwrapping the optional is that more or less accurate?